state-sync/src/request_manager.rs

```diff
@@ -295,10 +296,7 @@ impl RequestManager {
         let mut failed_peer_sends = vec![];
 
         for peer in peers {
-            let sender = self
-                .network_senders
-                .get_mut(&peer.network_id())
-                .expect("missing network sender for peer");
+            let mut sender = self.get_network_sender(&peer);
```

Comment:
Much better, I saw similar things in mempool where it had oh, let me crash cause it's not in the map and then all over the place